x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/SurrogateExpression.java

```diff
@@ -15,5 +15,8 @@
  */
 public interface SurrogateExpression {
 
+    /**
+     * @return the expression to be replaced by or {@code null} if this cannot be replaced
+     */
     Expression surrogate();
```

Comment:
As we don't generate any javadocs for the code, no need to use the @param/@return code which decrease readability - instead just add the comment as you normally would.
```
/**
 * Returns the replacing surrogate expression or null, if replacement is not possible.
 */
```